timeline-chart/src/time-graph-model.ts

```diff
@@ -40,6 +40,13 @@ export namespace TimelineChart {
         readonly data?: { [key: string]: any }
     }
 
+    export interface TimeGraphMarkerState extends TimeGraphState {
```

Comment:
The disadvantage of storing that you need to store next & prev markers in each marker after fetching the marker data which is unnecessary processing. Unnecessary, because users won't navigate through most of them, only some of them. On demand searching for the next marker by looping through all markers of a row should be acceptable.

Reply:
On demand searching implemented.